### PR TITLE
refactor: create and edit concern, file upload

### PR DESCRIPTION
**Pull request by omarCreativeDev**
TISNEW-5042

refactor: create and edit concern, file upload

- ensure on edit concern scenario form steps are pre populated including admin autocomplete component
- massage BE data so its consistent throught FE state
- added `PrepareUpload` state event due to flow changes - i.e add a concern api needs to be called before files can be uploaded to a concern
- updated concernId data type & cleaned up interfaces
- updated unit tests

### CODE REVIEW

File: src/app/concern/file-uploader/file-uploader.component.ts [hunk 1]
```diff
@@ -60,33 +60,13 @@ export class FileUploaderComponent implements OnInit, OnDestroy {
...
-          const files = filesInProgress.map(
-            (val: IFileUploadProgress) => val.file
-          );
-          this.upload(files);
-        }
-      })
-    );
-  }
-
+  // TODO check if needed why?
```
Comment (ojemuyiwa) [inline src/app/concern/file-uploader/file-uploader.component.ts:69]: when there's a concern ID the file upload progress will resume here.

File: src/app/concern/file-uploader/file-uploader.component.ts [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+  ViewChild
 } from "@angular/core";
 import { FormBuilder, FormGroup } from "@angular/forms";
 import { Select, Store } from "@ngxs/store";
 import { Observable, Subscription } from "rxjs";
 import { SnackBarService } from "../../shared/services/snack-bar/snack-bar.service";
+import { IConcernSummary, IFileUploadProgress } from "../concern.interfaces";
 import { ACCEPTED_IMAGE_MIMES } from "../constants";
-import { Upload } from "../state/concern.actions";
+import { PrepareUpload } from "../state/concern.actions";
```
Comment (ojemuyiwa) [inline src/app/concern/file-uploader/file-uploader.component.ts:14]: This functionality was already written. Files In Progress already Means Files are in progress.

There was listener on selectedConcern which would execute the upload when an id is returned. I.e. concern has been saved.

File: src/app/concern/file-uploader/file-uploader.component.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/concern/file-uploader/file-uploader.component.ts:69]: no worries feel free to update

File: src/app/concern/file-uploader/file-uploader.component.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/concern/file-uploader/file-uploader.component.ts:14]: no worries feel free to update

**Review by ojemuyiwa — APPROVED** [review]

**Review by ojemuyiwa — APPROVED** [review]

**Pull request merged**